jax/numpy/lax_numpy.py

```diff
@@ -2761,7 +2755,7 @@ def _index_to_gather(x_shape, idx):
   collapsed_slice_dims = []
   start_index_map = []
 
-  gather_indices = zeros((0,), dtype=int32)
+  gather_indices = onp.zeros((0,), dtype=int32)
```

Comment:
Why this change? Revert?

Reply:
It was causing an extra pointless compile. I'll check if it still is.

Reply:
Yeah, it can cause us to compile and execute an extra `lax.broadcast_in_dim` versus just doing a device transfer the first time this value is consumed (and we still have to do the same sized device transfer when we execute the broadcast_in_dim). There's no downside to using `onp.zeros` here.

Reply:
It's worth a comment then so no-one accidentally reverts it.

(Although really this whole function should be under `jit` and this should be staged into that `jit` computation.)

Reply:
WDYT?

Reply:
Oops, my browser was out of sync and so my "WDYT?" was mistimed.

Will comment!

Reply:
Done.

Agree that this whole function, as with others, should be under `jit`, though I'm hesitant to do that without even some rudimentary performance regression tests.